### ISSUE TITLE
Live view aspect ratio different than original camera stream

### ISSUE THREAD

**Issue by rici44** [AlexxIT/go2rtc#1466]
I setup Frigate v14 in privileged container on Raspberry Pi CM4 module and connected Tapo C120 camera as tapo source. Frigate automatically choose MSE/WebRTC streams which are provided by go2rtc v1.9.2. 

If MSE stream  is chosen then live view aspect ratio is different than original camera stream. [Camera](https://www.tp-link.com/us/home-networking/cloud-camera/tapo-c120/#specifications) has stream aspect ratio 16:9 but go2rt v1.9.2 provide MSE stream in aspect ratio 4:3.

If WebRTC stream is chosen then live aspect ratio is the same as the original camera stream 16:9. 

I tested standalone go2rtc v1.9.7 with the same config of the same camera and results are the same (MSE 4:3, WebRTC 16:9).

**Is it possible to setup correct aspect ratio in MSE stream ?**
**Is go2rtc doing something with original stream when providing MSE stream?**

**Comment by AlexxIT** [comment]
The camera is probably wrongly sending this information about the ratio. It's a fairly common situation.
Unfortunately I don't have any cameras with these problems to fix.
You can try to use ffmpeg source.

**Comment by rici44** [comment]
> The camera is probably wrongly sending this information about the ratio. It's a fairly common situation.

Why MSE and WebRTC streams are both OK in 16:9 when using Firefox browser?

> Unfortunately I don't have any cameras with these problems to fix.

What would help to solve this exact camera?

> You can try to use ffmpeg source

Won't it get more resources from CM4?

**Comment by dgibbs64** [comment]
Yeah, im getting the impression that just using rtsp might be the best option. I think 2-way audio might be the only thing I lose.

**Comment by adamoutler** [comment]
As a workaround to this go2rtc issue, I found the least invasive way to pre-process the stream is to add an ffmpeg compatibility layer as follows:
```
go2rtc:
  streams:
    driveway_stream: onvif://admin:admin@10.10.187.108:8000?subtype=CH01
    driveway_stream_ffmpeg: ffmpeg:rtsp://127.0.0.1:8554/driveway_stream#video=copy
```
This allows for simply changing the reference back to the source at a later time rather than working out new URLs and changing the original source.

**Comment by AlexxIT** [comment]
You right. FFmpeg can fix broken metadata. But it increases the start time of the stream.

**Comment by lngarrett** [comment]
Hi, I'm very new to go2rtc so I might be misunderstanding. Is this work around meant to preserve 2-way audio and also fix aspect ratio? I'm trying to achieve both, but the work around seems to break 2-way audio.

```
foocam_tapo:
      - tapo://admin:sha256@foocam.whisk.ee
foocam:
      - ffmpeg:foocam_tapo
```

**Comment by AlexxIT** [comment]
```yaml
streams:
  tapo:
  - ffmpeg:rtsp://...
  - tapo://...
```

**Comment by NevarVit** [comment]
On cameras with OpenIPC firmware there is the same problem :(

**Comment by borgqueenx** [comment]
Was there any luck fixing this issue?
